### PR TITLE
feat: RSDKUtils macOS Support (SDKCF-5397)

### PR DESCRIPTION
**Pull request by malayli**
## Description
RSDKUtils macOS Support for the macOS PushTester app.

### CODE REVIEW

File: Tests/Tests/UIColorExtensionsSpec.swift [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#if os(iOS) || os(watchOS)
```
Comment (malayli) [inline Tests/Tests/UIColorExtensionsSpec.swift:1]: UIColor is available on WatchOS:
https://developer.apple.com/documentation/uikit/uicolor

File: README.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ RSDKUtils consists of 4 sub-modules:
 
 This module supports iOS 12.0 and above. It has been tested with iOS 12.0 and above.
 This module also supports watchOS 6.0 and above, and it's compatible with extensions.
+This module also supports macOS 11.0 and above.
```
Comment (imaginaris) [inline README.md:17]: Maybe let's combine those 3 sentences about supported OSes. I'm not good at it but maybe something like:

This module supports following operating systems:
* iOS 12.0 and above (including extension contexts)
* watchOS 6.0 and above
* macOS 11.0 and above

It has been tested with iOS 12.0 and above.

File: Sources/RSDKUtilsMain/Extensions/UIColor+Extensions.swift [hunk 3]
```diff
@@ -1,3 +1,4 @@
+#if os(iOS)
```
Comment (imaginaris) [inline Sources/RSDKUtilsMain/Extensions/UIColor+Extensions.swift:1]: You said that UIColor was available on watchOS some maybe let's add  `|| os(watchOS)` here 🙂

**Review by imaginaris — COMMENTED** [review]
Nice! 
Just 2 comments from me.

**Comment by apparition47** [comment]
If this change (for macOS 11+) is for the push tester app (10.14), will we have to bump its min target too?

**Review by imaginaris — APPROVED** [review]

**Review by apparition47 — APPROVED** [review]

**Comment by malayli** [comment]
> If this change (for macOS 11+) is for the push tester app (10.14), will we have to bump its min target too?

Actually PushTester app deployment target is 10.13.
Then I changed the version to `.macOS(.v10_13)` in Package.swift

**Pull request merged**